### ISSUE TITLE
ModuleNotFoundError: No module named 'fb' when running NMT training script

### ISSUE THREAD

**Issue by KonceptBlast** [facebookresearch/UnsupervisedMT#38]
While trying to run the training script `python3 main.py` I receive the following error:
>   File "<my_project_directory>/NMT/src/data/loader.py", line 164, in load_para_data
    data1 = load_binarized(path.replace('XX', lang1), params)
  File "<my_project_directory>/NMT/src/data/loader.py", line 32, in load_binarized
    data = torch.load(path)
  File "<my_virtualenv_directory>/lib/python3.6/site-packages/torch/serialization.py", line 358, in load
    return _load(f, map_location, pickle_module)
  File "<my_virtualenv_directory>/lib/python3.6/site-packages/torch/serialization.py", line 542, in _load
    result = unpickler.load()
ModuleNotFoundError: No module named 'fb'

It seems that there is some dependency missing that I'm not aware of.
I use Ubuntu 18.04.1, Python 3.6.6  and PyTorch 0.4.1

**Comment by glample** [comment]
Hi,

This is weird, I'm not aware of such package. What is the full command you used to run the model?

**Comment by KonceptBlast** [comment]
Hi, 
given that I run the command from NMT directory, the command looks as follows:

> python3 main.py --exp_name transformer-deen --transformer True --n_enc_layers 4 --n_dec_layers 4 --share_enc 3 --share_dec 3 --share_lang_emb True --share_output_emb True --langs 'de,en' --n_mono -1 --mono_dataset 'de:../../data/training.de.tok.50000.pth,,;en:../../data/training.en.tok.50000.pth,,' --para_dataset 'de-en:,../../data/dev.tok.XX.50000.pth,../../data/test.tok.XX.50000.pth' --mono_directions 'de,en' --word_shuffle 3 --word_dropout 0.1 --word_blank 0.2 --pivo_directions 'de-en-de,en-de-en' --pretrained_emb '../../data/training.de-en.50000.vec' --pretrained_out True --lambda_xe_mono '0:1,100000:0.1,300000:0' --lambda_xe_otfd 1 --otf_num_processes 30 --otf_sync_params_every 1000 --enc_optimizer adam,lr=0.0001 --epoch_size 500000 --stopping_criterion bleu_cs_pl_valid,10

**Comment by glample** [comment]
This looks right. Did you make any changes to the code of the original repository?
Also, do you get the same error when reloading any of the binarized files?

**Comment by KonceptBlast** [comment]
I didn't introduce any changes to the original code (PyTorch model). I only changed `get_data.sh` script to download and preprocess my datasets, but it seemed to worked fine because it generated *.pth files as well as one binary file `training.de-en.50000.bin`

> Also, do you get the same error when reloading any of the binarized files?
How would you like me to test it?

**Comment by glample** [comment]
You can try to open python in the main directory, `import torch`, and then try `torch.load(path)` where `path` is the `path` of the different `.pth` files.

**Comment by KonceptBlast** [comment]
I tried and it gives the same error message.
To compare if it happens with other serialized models, I loaded some `.pt` file from my disk and it didn't return error.
So I guess it seems that `.pth` files generated during preprocessing part are corrupted? If so, how should I investigate the problem?
Thanks for your help.

**Comment by glample** [comment]
can you maybe just re-run the binarizing step very quickly and see if you still have the same error? Like, remove the .pth and rerun preprocess.py
If this still does not work, you can send me the .pth of the test set which is small, and I can try to reload it.

**Comment by KonceptBlast** [comment]
I did as you suggested and preprocessed files again. Now it seems that newly generated `.pth` files work :+1: 
I checked out what was written in old and new log files, and in both cases summary infos (# words etc.) were the same. 
So I checked the size of old and new `.pth` files and it appears that there is difference - all older `.pth` files have 3 bytes more ... I will close this issue, but it is rather interesting why old `.pth` files were corrupted.
